pymongo/asynchronous/database.py

```diff
@@ -119,9 +119,16 @@ def __init__(
             read_concern or client.read_concern,
         )
 
+        from pymongo.asynchronous.mongo_client import AsyncMongoClient
+
         if not isinstance(name, str):
             raise TypeError("name must be an instance of str")
 
+        if not isinstance(client, AsyncMongoClient):
+            raise TypeError(
+                f"AsyncMongoClient required but {client} is an instance of {type(client)}"
```

Comment:
Do we need to include both the repr(client) and the type(client)? It could be safer to only add the type to avoid adding topology/hostname info into these errors.